src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs

```diff
@@ -257,9 +257,7 @@ public bool TryWriteProperties(PropertiesWriter writer, IList<ProjectInfo> proje
     /// </summary>
     public DirectoryInfo ComputeProjectBaseDir(IList<DirectoryInfo> projectPaths)
     {
-        var projectBaseDir = analysisConfig.LocalSettings
-            ?.Find(x => ConfigSetting.SettingKeyComparer.Equals(SonarProperties.ProjectBaseDir, x.Id))
-            ?.Value;
+        var projectBaseDir = analysisConfig.GetSettingOrDefault(SonarProperties.ProjectBaseDir, includeServerSettings: false, defaultValue: null, logger);
```

Comment:
This lovely `GetSettingOrDefault` checks everything, not only the analysis config file.
Specifically, it checks, in this order (which is the same as the begin steps' order):
- LocalSettings
- GlobalFile
- ScannerEnv
- ServerSettings

I can't think of a way this will break, as we are always respecting the user's decision with this method.

Reply:
I see that `includeServerSettings`  is set to false. Is there a reason to exclude those? Can't this property be set in the server configuration?

Reply:
good point, I added supported for server properties.